lib/build-cloud/launchconfiguration.rb

```diff
@@ -80,8 +77,132 @@ def create
+            munged_options = options.dup
+            
+            # default ebs_optimized is false
+            if munged_options[:ebs_optimized].nil?
+                munged_options[:ebs_optimized] = false
+            end
+            
+            differences = Hash[*(
+                (fog_options.size > munged_options.size)    \
+                  ? fog_options.to_a - munged_options.to_a \
+                  : munged_options.to_a - fog_options.to_a
+                ).flatten] 
+                
+            @log.debug("Fog options: #{fog_options.inspect}")
+            @log.debug("Munged options: #{munged_options.inspect}")
+            
+            unless fog_options == munged_options
+                @log.debug("Differences between fog and options is: #{differences}")
+                
+                @log.info("Updating Launch Configuration #{fog_object.id}")
+                # Now for some useful messaging
+                differences.each do |k,v|
+                    @log.info(" ... updating #{k}")
+                end
+                
+                # create a temp configuration
+                fog_options[:id] = "#{fog_options[:id]}_temp"
+                
+                dsl = @as.describe_launch_configurations( { 'LaunchConfigurationNames' => fog_options[:id]} )
+                dsl = dsl.body['DescribeLaunchConfigurationsResult']['LaunchConfigurations']
+                if dsl.empty?
+                    @log.info( "Creating launch configuration #{fog_options[:id]}" )
+                    
+                    temp_launch_config = @as.configurations.new( fog_options )
+                    temp_launch_config.save
+                    
+                    # update any associated ASG's
+                    # get list of ASG's
+                    asgs = @as.describe_auto_scaling_groups().body['DescribeAutoScalingGroupsResult']['AutoScalingGroups']
+                    
+                    # Update any that are using our taget LaunchConfigurationName
+                    asgs.each do |asg|
+                        @log.debug("Checking ASG #{asg['AutoScalingGroupName']}")
+                        if asg['LaunchConfigurationName'] == options[:id]
+                            @log.info("Updating ASG #{asg['AutoScalingGroupName']} ")
+                            @as.update_auto_scaling_group(asg['AutoScalingGroupName'], {'LaunchConfigurationName' => fog_options[:id]})
+                        end
+                    end
+                    
+                    # delete current ASG Configuration
+                    @as.delete_launch_configuration( options[:id] )
+                    
+                    # create new fog object
+                    @log.debug( "Updating launch configuration #{options[:id]}" )
+                    launch_config = @as.configurations.new( options )
+                    launch_config.save
+                    
```

Comment:
If we get failures in the preceding lines, it reads like we'd end up with temporary LCs mapped to ASGs, and subsequent runs of build-cloud would then fail, is that correct?  Can we make this safer?

When I was thinking over this lifecycling, I wondered about timestamping the names of the LCs, and then always mapping newer LC names to the ASGs (and deleting old, unused ones).  I don't know if that's the right approach, but it might help this failure case.